src/v/cloud_storage_clients/abs_client.cc

```diff
@@ -374,8 +374,19 @@ abs_client::abs_client(
 }
 
 ss::future<client_self_configuration_result> abs_client::self_configure() {
-    // TODO: A future commit will plug in the implementation
-    co_return abs_self_configuration_result{.is_hns_enabled = false};
+    auto result = co_await get_account_info(5s);
```

Comment:
Should this be configurably retried indefinitely (or with some higher timeout, 5s seems a bit low)? Seems a bit fishy to silently assume HNS is disabled, especially if it means littering directories

Reply:
I see. Maybe we should fail the process if this fails then? It seems like something we'd see in an incident, where we'd see disks filling up for no good reason, only to realize this check assumed HNS is disabled

Reply:
I'm not against that, I guess I could wire the top level `stop_signal` to the client pool and use that to gracefully stop. This means that we would also stop on any misconfiguration, but I think that's actually desirable.

Reply:
Implemented it. I lumped it all into one commit at the end because it felt a bit intrusive. Let me know how you feel about it.